_posts/2016-11-17-JAX-London-2016.md

```diff
@@ -0,0 +1,691 @@
+It’s awesome when enterprises contribute to open source.
+
+- Bosch: They are doing interesting work with Eclipse Foundation, they know they need to do open source so they make open source contributions
+- Comcast: Everybody hates Comcast in the US, but they are making open source contributions
+
+James mentions a couple of things he finds important:
+
+- Microservices and container based deployment such as Docker and Kubernetes are very hot topics
+- Break down the technical model and teams
+- Continuous integration is very important, there are still people not using it!
+- Make people responsible for their on Quality Assurance
+- Embrace failure and graceful degradation
+
+Finally, he mentioned that there is always the need to deal with the politics and that governance is still important and needed.
+Oracle needs to give their commitment to Java and the Cloud and we as a community need to encourage Oracle to step up and not only complain about them.
+
+Links:
+
+- [Presentation](http://www.slideshare.net/monkchips/java-for-cloud-natives-at-jax-london-2016)
+
+## Developing Microservices with aggregates - Chris Richardson
+
+<img class="float-image p-image" src="{{ '/img/jax-london-2016/chris-richardson.jpg' | prepend: site.baseurl }}" alt="Chris Richardson" style="width: 100px"/>
+
+The goal of [Chris Richardson](https://twitter.com/crichardson)'s presentation was to show how Domain-Driven Design aggregates and microservices are a perfect match.
+
+A microservice based architecture tackles complexity through modularisation.
+A microservice should be seen as a business capability for example a catalog service, review service or order service.
+By having service boundaries you enforce modularity.
+Also important to mention is that each microservice needs to have its own database, microservices do not share a database!
+Finally there should always be an API gateway in front of the microservices, which is the entrypoint for the frontend user interface and mobiles devices.
+They should never access your microservice directly!
+
+Chris strongly suggest reading [Domain Driven Design](https://www.amazon.com/Domain-Driven-Design-Tackling-Complexity-Software/dp/0321125215) by Evan Evans.
+The core building blocks of Domain-Driven Design are the following:
+
+- Entities
+- Value objects
+- Services
+- Repositories
+- Aggregates
+
+About [Domain-Driven Design aggregates](http://martinfowler.com/bliki/DDD_Aggregate.html):
+
+- Cluster of objects that can be treated as a unit
+- Graph consisting of a root entity and one or more other entities and value objects
+- Typically business entities are aggregates e.g., Customer, Account, Order, Product
+
+Chris talked about the problems you have to deal with when dealing with microservices and that practicing Domain-Driven Design can help you a lot to address these.
```

Comment:
you have to deal with when dealing with ... lot's of with's.

maybe replace to you have to deal with by you have to handle